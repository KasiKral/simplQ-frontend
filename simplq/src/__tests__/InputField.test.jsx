import React from 'react';
import { render, fireEvent, cleanup } from '@testing-library/react';
import '@testing-library/jest-dom';
import InputField from '../components/common/InputField';

afterEach(cleanup);

it('renders correctly', () => {
  const { asFragment } = render(<InputField />);
  expect(asFragment()).toMatchSnapshot();
});

it('Name placholder change', () => {
  const placeholderText = 'UserInput';
  let userInput = 'Input';
  const { getByPlaceholderText } = render(<InputField placeholder={placeholderText} />);
  fireEvent.change(getByPlaceholderText(placeholderText), { target: { value: userInput } });
  expect(getByPlaceholderText(placeholderText)).toHaveValue(userInput);
});
