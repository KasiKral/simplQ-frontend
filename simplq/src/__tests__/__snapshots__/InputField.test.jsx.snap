// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<DocumentFragment>
  <div
    class="MuiFormControl-root MuiTextField-root MuiFormControl-fullWidth"
  >
    <div
      class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-fullWidth MuiInputBase-formControl"
    >
      <input
        aria-invalid="false"
        class="MuiInputBase-input MuiOutlinedInput-input"
        required=""
        type="text"
        value=""
      />
      <fieldset
        aria-hidden="true"
        class="PrivateNotchedOutline-root-1 MuiOutlinedInput-notchedOutline"
        style="padding-left: 8px;"
      >
        <legend
          class="PrivateNotchedOutline-legend-2"
          style="width: 0.01px;"
        >
          <span>
            â€‹
          </span>
        </legend>
      </fieldset>
    </div>
  </div>
</DocumentFragment>
`;
